src/client/oicp/CpoOICPClient.ts

```diff
@@ -0,0 +1,1303 @@
+        detailedMessages: { logs: sendResult.logs },
+        module: MODULE_NAME, method: 'sendEvseStatuses'
+      });
+    }
+    // Save result in oicp endpoint
+    this.oicpEndpoint.lastPatchJobOn = startDate;
+    // Set result
+    if (sendResult) {
+      this.oicpEndpoint.lastPatchJobResult = {
+        successNbr: sendResult.success,
+        failureNbr: sendResult.failure,
+        totalNbr: sendResult.total,
+        chargeBoxIDsInFailure: _.uniq(sendResult.chargeBoxIDsInFailure),
+        chargeBoxIDsInSuccess: _.uniq(sendResult.chargeBoxIDsInSuccess)
+      };
+    } else {
+      this.oicpEndpoint.lastPatchJobResult = {
+        successNbr: 0,
+        failureNbr: 0,
+        totalNbr: 0,
+        chargeBoxIDsInFailure: [],
+        chargeBoxIDsInSuccess: []
+      };
+    }
+    // Save
+    await OICPEndpointStorage.saveOicpEndpoint(this.tenant.id, this.oicpEndpoint);
+    // Return result
+    return sendResult;
+  }
+
+  /**
+   * Push EVSE
+   */
+  async pushEvseData(evses: OICPEvseDataRecord[], actionType: OICPActionType): Promise<OICPAcknowledgment> {
+    let pushEvseDataResponse: OICPAcknowledgment = {} as OICPAcknowledgment;
+    // Check for input parameter
+    if (!evses) {
+      throw new BackendError({
+        action: ServerAction.OICP_PUSH_EVSES,
+        message: 'Invalid parameters',
+        module: MODULE_NAME, method: 'pushEvseData',
+      });
+    }
+    // Get EVSE endpoint url
+    // for QA - environment: https://service-qa.hubject.com/api/oicp/evsepush/v23/operators/{operatorID}/data-records
+    // for PROD - environment: https://service.hubject.com/api/oicp/evsepush/v23/operators/{operatorID}/data-records
+    const fullUrl = this.getEndpointUrl('evses', ServerAction.OICP_PUSH_EVSES);
+
+    // Build payload
+    const operatorEvseData: OICPOperatorEvseData = {} as OICPOperatorEvseData;
+    operatorEvseData.OperatorID = this.getOperatorID(ServerAction.OICP_PUSH_EVSES);
+    operatorEvseData.OperatorName = this.tenant.name;
+    operatorEvseData.EvseDataRecord = evses;
+
+    const payload: OICPPushEvseDataCpoSend = {} as OICPPushEvseDataCpoSend;
+    payload.ActionType = actionType;
+    payload.OperatorEvseData = operatorEvseData;
+
+    // Log
+    Logging.logDebug({
+      tenantID: this.tenant.id,
+      action: ServerAction.OICP_PUSH_EVSES,
+      message: `Push EVSEs from tenant: ${this.tenant.id}`,
+      module: MODULE_NAME, method: 'pushEvseData',
+      detailedMessages: { payload }
+    });
+
+    const publicCert = this.getClientCertificate(ServerAction.OICP_PUSH_EVSES);
+    const privateKey = this.getPrivateKey(ServerAction.OICP_PUSH_EVSES);
+
+    const httpsAgent = new https.Agent({
+      rejectUnauthorized: false, // (NOTE: this will disable client verification)
+      cert: publicCert,
+      key: privateKey,
+      passphrase: ''
+    });
+
+    // Call Hubject
+    await axios.post(fullUrl, payload, {
```

Comment:
Please the axios factory to create axios client everywhere.

Reply:
okay